src/library/dao/playlistdao.cpp

```diff
@@ -397,24 +397,18 @@ void PlaylistDAO::removeHiddenTracks(const int playlistId) {
     ScopedTransaction transaction(m_database);
     // This query deletes all tracks marked as deleted and all
     // phantom track_ids with no match in the library table
-    QString queryString =
-            QStringLiteral(
-                    "SELECT position FROM PlaylistTracks "
-                    "WHERE PlaylistTracks.id NOT IN ("
-                    "SELECT PlaylistTracks.id FROM PlaylistTracks "
-                    "INNER JOIN library ON library.id = PlaylistTracks.track_id "
-                    "WHERE PlaylistTracks.playlist_id = %1 "
-                    "AND library.mixxx_deleted = 0 ) "
-                    "AND PlaylistTracks.playlist_id = %1")
-                    .arg(playlistId);
-
     QSqlQuery query(m_database);
-    if (!query.prepare(queryString)) {
-        LOG_FAILED_QUERY(query);
-        return;
-    }
-
+    query.prepare(QStringLiteral(
+            "SELECT p1.position FROM PlaylistTracks p1"
+            "WHERE p1.id NOT IN ("
+            "SELECT p2.id FROM PlaylistTracks p2"
+            "INNER JOIN library ON library.id = p2.track_id "
+            "WHERE p2.playlist_id = p1.playlist_id "
+            "AND library.mixxx_deleted = 0 ) "
+            "AND p1.playlist_id=:id"));
```

Comment:
No super sure about this one, please verify if this is correct.

Reply:
I'll check and test this later.